Userland/Libraries/LibC/signal.cpp

```diff
@@ -252,6 +252,9 @@ int getsignalbyname(char const* name)
     VERIFY(name);
     StringView name_sv { name, strlen(name) };
     for (size_t i = 0; i < NSIG; ++i) {
+        if (!sys_signame[i])
+            continue;
```

Comment:
I was confused at first by the `sys_signame` array in that `SIGCANCEL` was missing but the `static_assert` was not triggering; saw the definition in the `.h` later on. Could you add a comment for index 32 in that array to emphasize the fact that we left it out on purpose?

Reply:
Yeah, I got confused by the same thing. I think a similar comment in `sys_siglist` would also be helpful, as well as a similar null check in `strsignal()`, which returns `null` for SIGCANCEL at the moment.

I'm not sure we should be treating `0` as a valid signum in `getsignalbyname()`, `getsignalname()` and `strsignal()` either.

It might even be worth initializing both these arrays with the preprocessor in a way similar to how we enumerate parser tokens., i.e:

```c++
#define ENUMERATE_SIGNALS                 \
    __ENUMERATE_SIGNAL(1, "HUP", "Hangup") \
    __ENUMERATE_SIGNAL(2, "INT", "Interrupt")
   etc..
```

Reply:
Definitely sounds like an improvement!